File: app/src/main/java/com/example/mvvmgithubsampleapp/di/AppModule.java
package com.example.mvvmgithubsampleapp.di;

import com.example.mvvmgithubsampleapp.repository.ProjectService;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

@Module
class AppModule {
    @Singleton @Provides
    ProjectService provideGithubService() {
        return new Retrofit.Builder()
                .baseUrl("https://api.github.com/")
                .addConverterFactory(GsonConverterFactory.create())
                .build()
                .create(ProjectService.class);
    }
}

File: app/src/main/java/com/example/mvvmgithubsampleapp/di/AppModule.kt
package com.example.mvvmgithubsampleapp.di

import com.example.mvvmgithubsampleapp.repository.ProjectService
import com.jakewharton.retrofit2.adapter.kotlin.coroutines.CoroutineCallAdapterFactory
import dagger.Module
import dagger.Provides
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Singleton

@Module
internal class AppModule {
    @Singleton
    @Provides
    fun provideGithubService(): ProjectService {
        return Retrofit.Builder()
            .baseUrl("https://api.github.com/")
            .addConverterFactory(GsonConverterFactory.create())
            .addCallAdapterFactory(CoroutineCallAdapterFactory())
            .build()
            .create(ProjectService::class.java)
    }
}